internal/controller/statusreport/statusreport_adapter.go

```diff
@@ -238,3 +235,108 @@ func (a *Adapter) findUntriggeredIntegrationTestFromStatus(integrationTestScenar
+		srs, _ = status.NewSnapshotReportStatus("")
+	}
+
+	// Report the integration test status to pr/commit included in the tested component snapshot
+	// or the component snapshot included in group snapshot
+	for _, destinationComponentSnapshot := range destinationSnapshots {
+		destinationComponentSnapshot := destinationComponentSnapshot
+		reporter := a.status.GetReporter(destinationComponentSnapshot)
+		if reporter == nil {
+			a.logger.Info("No suitable reporter found, skipping report")
+			continue
+		}
+		a.logger.Info(fmt.Sprintf("Detected reporter: %s", reporter.GetReporterName()))
+
+		if err := reporter.Initialize(ctx, destinationComponentSnapshot); err != nil {
+			a.logger.Error(err, "Failed to initialize reporter", "reporter", reporter.GetReporterName())
+			return fmt.Errorf("failed to initialize reporter: %w", err)
+		}
+		a.logger.Info("Reporter initialized", "reporter", reporter.GetReporterName())
+
+		// set componentName to component name of component snapshot or pr group name of group snapshot when reporting status to git provider
+		componentName := ""
+		if gitops.IsGroupSnapshot(testedSnapshot) {
+			componentName = "pr group " + testedSnapshot.Annotations[gitops.PRGroupAnnotation]
+		} else if gitops.IsComponentSnapshot(testedSnapshot) {
+			componentName = testedSnapshot.Labels[gitops.SnapshotComponentLabel]
+		}
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			for _, integrationTestStatusDetail := range integrationTestStatusDetails {
+				if srs.IsNewer(integrationTestStatusDetail.ScenarioName, integrationTestStatusDetail.LastUpdateTime) {
+					a.logger.Info("Integration Test contains new status updates", "scenario.Name", integrationTestStatusDetail.ScenarioName)
+				} else {
+					//integration test contains no changes
+					continue
+				}
+				testReport, reportErr := status.GenerateTestReport(ctx, a.client, *integrationTestStatusDetail, testedSnapshot, componentName)
+				if reportErr != nil {
+					if writeErr := status.WriteSnapshotReportStatus(ctx, a.client, testedSnapshot, srs); writeErr != nil { // try to write what was already written
+						return fmt.Errorf("failed to generate test report AND write snapshot report status metadata: %w", e.Join(reportErr, writeErr))
+					}
+					return fmt.Errorf("failed to generate test report: %w", reportErr)
+				}
+				if reportStatusErr := reporter.ReportStatus(ctx, *testReport); reportStatusErr != nil {
+					if writeErr := status.WriteSnapshotReportStatus(ctx, a.client, testedSnapshot, srs); writeErr != nil { // try to write what was already written
+						return fmt.Errorf("failed to report status AND write snapshot report status metadata: %w", e.Join(reportStatusErr, writeErr))
+					}
+					return fmt.Errorf("failed to update status: %w", reportStatusErr)
+				}
+				srs.SetLastUpdateTime(integrationTestStatusDetail.ScenarioName, integrationTestStatusDetail.LastUpdateTime)
```

Comment:
Maybe we can extract this logic to a separate function that creates and updates the integration report status to reduce the cognitive complexity for `ReportSnapshotStatus` (currently 20).

Reply:
Thanks. I will work on it.

Reply:
Updated.